### ISSUE TITLE
Unbound rejects requests when SNAT_TO_SOURCE is configured

### ISSUE THREAD

**Issue by slintes** [mailcow/mailcow-dockerized#3505]
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [x] I understand, that not following or deleting the below instructions, will result in immediate closing and deletion of my issue.
- [x] I have understood that answers are voluntary and community-driven, and not commercial support.
- [x] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

**Description of the bug**:
- I was running mailcow-dockerized from 2019-12-11-22-32-09
- on Saturday, 2020-04-25, the server rebooted
- all incoming emails were rejected after that, because unbound did not answer DNS request of the postfix container anymore
- updating mailcow to 2020-04-26-21-11-44 didn't help
- adding this line to the unbound config finally solved the problem
`access-control: *my-snat-to-source-ip* allow`

So basically this throws 2 questions:
- why did a server restart change behaviour? (is `latest` used somewhere for a container image, or any other moving tag?)
- should mailcow configure unbound to allow the SNAT_TO_SOURCE IP, or did I miss something?

**Docker container logs of affected containers**:
Unfortunately I don't have the logs anymore. Out of my head:
- postfix was complaining about something like "Sender rejected, unknown Domain"
- DNS resolution on the host was ok
- DNS resolution from inside the unbound container was ok
- after increasing verbosity in unbound I saw lots of something like "request from *my-snat-to-source-ip* rejected"

**Reproduction of said bug**:
nothing to add here

**System information**:

| Question | Answer |
| --- | --- |
| My operating system | CentOS 7 |
| Is Apparmor, SELinux or similar active? | no |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | it is a Hetzner Cloud server, IIRC they use KVM |
| Server/VM specifications (Memory, CPU Cores) | not relevant |
| Docker Version (`docker version`) | updated to 19.03.8 during debugging the issue |
| Docker-Compose Version (`docker-compose version`) | 1.25.5 |
| Reverse proxy (custom solution) | n/a |

- Output of `git diff origin/master`, any other changes to the code? If so, **please post them**.

Nothing relevant IMHO, but here you go: https://pastebin.com/PV0aA2Wf

- All third-party firewalls and custom iptables rules are unsupported. *Please check the Docker docs about how to use Docker with your own ruleset*. Nevertheless, iptabels output can help us to help you: `iptables -L -vn`, `ip6tables -L -vn`, `iptables -L -vn -t nat` and `ip6tables -L -vn -t nat`.

https://pastebin.com/fikhrPy2

- DNS problems? Please run `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @172.22.1.254` (set the IP accordingly, if you changed the internal mailcow network) and post the output.

Didn't work while having the issue, works now (returning the SO IPs)

**Comment by andryyy** [comment]
That's strange, cannot confirm it on any system with SNAT configured.

I will dig into it. Sounds interesting.

**Comment by slintes** [comment]
In case you have a good idea how to further debug this, I'm willing to temporarily revert my change and do some more investigation.

**Comment by andryyy** [comment]
It does not SNAT the destination address IPV4_NETWORK.

```
    rule = iptc.Rule()
    rule.src = os.getenv('IPV4_NETWORK', '172.22.1') + '.0/24'
    rule.dst = '!' + rule.src
    target = rule.create_target("SNAT")
    target.to_source = snat_target
```

**Comment by slintes** [comment]
I can't follow, sorry... is this a mailcow issue, or is my host misconfigured?

**Comment by andryyy** [comment]
I see this in your natting table:

```
62576 4317K SNAT       all  --  *      !mailcow  172.22.1.0/24        0.0.0.0/0            to:<my-snat-to-source-ip>
54218 4210K SNAT       all  --  *      *       172.22.1.0/24       !172.22.1.0/24        to:<my-snat-to-source-ip>
```

The second rule is "ours", the first is not.

Try:

```
iptables -L -vn -t nat --line-numbers
# Get the line number and delete it (for example nr. 123)
iptables -t nat -D POSTROUTING 123
```

**Comment by slintes** [comment]
Yes, that's it!
I created that rule (and a boot script for it, and forgot about it) almost 2 years ago... IIRC mailcow did not support this itself at that time...? So disabling that boot script and a reboot did the trick, then I reverted unbound config and restarted it, and it looks good.
Just for completeness: the 1st rule was mine, the 2nd yours. At least I see the 2nd only now.

Thanks a lot for your help! And for mailcow overall :+1: :slightly_smiling_face:
